### DISCUSSION TITLE
Warning: Prop id did not match. Server: "sas_78372_tTPoc3REu9g_NbK7AQf2B" Client: "sas_78372_zGToWwkfzFKlIF36qF6gB"

### DISCUSSION THREAD

**Discussion by obrunodelgado** [jaredpalmer/razzle#1656] [Q&A] [answered]
Hello,

I need to use a random ID for an HTML element in the DOM (rendered in server side) and when the component did mount I need to interact with this element using it's identifier.

The problem is that the code will generate different IDs in the front-end and in the back-end.

How can I share a backend generated value with the frontend of a component that should appear more than once in the dom with different ids?

Frontend Code:
![image](https://user-images.githubusercontent.com/2372140/124648586-599bbb80-de8f-11eb-922b-5b7a256dd9c4.png)

Backend Code:
![image](https://user-images.githubusercontent.com/2372140/124648600-602a3300-de8f-11eb-870c-d4fe1c8ed460.png)

In the nextJS I could use the _dynamic_ feature to solve the problem.

**Comment by fivethreeo** [comment]
Set the shortid on a script tag. E.g. window.shortId = '${shortId}'; then use window.shortId in frontend


If multiple times you somehow have to use a deterministic way to map tthe ids to the component. Maybe hash the props except shortId and use that to get to the shortId?

**Reply by obrunodelgado** [reply]
I will need to generate the identifier multiple times and I can't figure it out how to generate a deterministic identifier because all the components contains the same properties. Basically I have this properties:

- Site Id
- Network Id
- Format Id
- Target
- Tag ID 

The Tag ID is the only property that will change, because it should be unique for each HTML tag.

Even if I just define the random identifier in the backend rendering I see the Prop Id did not match. Pretty hard.

**Comment by fivethreeo** [comment]
Then put all props in a array in a script tag. Render each in a loop using window client side.

**Comment by fivethreeo** [comment, marked as answer]
https://github.com/thearnica/react-uid

This seems to be a solution to this problem
